server/src/models/Client.ts

```diff
@@ -0,0 +1,18 @@
+import { Schema, model } from 'mongoose';
+import { IClient } from '../interfaces/client.interface';
+
+const ClientSchema: Schema = new Schema(
+  {
+    email: { type: String, required: true, unique: true },
+    password: { type: String, required: true, minLength: 8, maxLength: 255 },
+    fullName: { type: String, required: true },
+  },
+  {
+    timestamps: true,
+    versionKey: false,
+  }
+);
+
+const ClientModel = model<IClient>('Client', ClientSchema);
+
+export default ClientModel;
```

Comment:
What is the different between a client and a staff?

Reply:
You are right, I am not sure if I want to add more attributes to the Staff or to the Client collection, so I decided to put it separately, but I could do just User model.

Reply:
I have removed client and staff, and I added user and role instead.